pkgs/tools/backup/dar/default.nix

```diff
@@ -1,41 +1,62 @@
-{ lib, gccStdenv, fetchurl
-, which
-, attr, e2fsprogs
-, curl, libargon2, librsync, libthreadar
-, gpgme, libgcrypt, openssl
-, bzip2, lz4, lzo, xz, zlib
-, CoreFoundation
+args @ {
+  lib,
+  stdenv,
+  llvmPackages_12, # Anything newer than 11
+  fetchzip,
+  which,
+  attr,
+  e2fsprogs,
+  curl,
+  libargon2,
+  librsync,
+  libthreadar,
+  gpgme,
+  libgcrypt,
+  openssl,
+  bzip2,
+  lz4,
+  lzo,
+  xz,
+  zlib,
+  CoreFoundation,
 }:
 
-with lib;
-
 let
-  # Fails to build with clang on Darwin:
+  # Fails to build with clang-11 on Darwin:
   # error: exception specification of overriding function is more lax than base version
-  stdenv = gccStdenv;
+  stdenv = if args.stdenv.isDarwin then llvmPackages_12.stdenv else args.stdenv;
```

Comment:
We can move this to `all-packages.nix`, i.e.

```
  dar = callPackage ../tools/backup/dar {
    stdenv = if stdenv.isDarwin then llvmPackages_12.stdenv else stdenv;
  };
```

Reply:
I know. I prefer to keep the logic in the packages' files because all-packages is a giant mess and I'd rather not add to that.  
I also think it's easy to forget that the `callPackage` site has any special logic in it because that's not visible from the files that are supposed to declare the derivation.